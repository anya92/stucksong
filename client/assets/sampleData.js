export const recentlyPlayedTracks = [
  {
    album: 'Americana',
    artist: 'The Offspring',
    id: '4EchqUKQ3qAQuRNKmeIpnf',
    image: 'https://i.scdn.co/image/c77a1f3a2de72c2a8fde481a016376ca8ef81b33',
    played_at: '2018-08-14T18:50:09.757Z',
    title: 'The Kids Aren\'t Alright',
    uri: 'spotify:track:4EchqUKQ3qAQuRNKmeIpnf',
  },
  {
    album: 'No Pads, No Helmets...Just Balls (15th Anniversary Tour Edition)',
    artist: 'Simple Plan',
    id: '6otiaV2fagE3s8IvP6WkwG',
    image: 'https://i.scdn.co/image/9f6a521519080242f2de1c2b8873eec5372f7a6a',
    played_at: '2018-08-14T18:46:04.608Z',
    title: 'I\'m Just A Kid',
    uri: 'spotify:track:6otiaV2fagE3s8IvP6WkwG',
  },
  {
    album: 'All Killer, No Filler',
    artist: 'Sum 41',
    id: '1HNE2PX70ztbEl6MLxrpNL',
    image: 'https://i.scdn.co/image/4fc4f0bb924338f85e0fe7c899cde32ec5e9f10d',
    played_at: '2018-08-14T18:42:32.417Z',
    title: 'In Too Deep',
    uri: 'spotify:track:1HNE2PX70ztbEl6MLxrpNL',
  },
];

export const topArtists = [
  {
    genres: [
      'dance pop',
      'folk-pop',
      'indie pop',
      'indie r&b',
      'indietronica',
      'metropopolis',
      'pop',
    ],
    id: '4Ui2kfOqGujY81UcPrb5KE',
    image: 'https://i.scdn.co/image/a20acc6267aec9d08aa32d05ae2fb8647943f81a',
    name: 'HAIM',
    uri: 'spotify:artist:4Ui2kfOqGujY81UcPrb5KE',
  },
  {
    genres: [
      'chamber pop',
      'indie folk',
      'shimmer pop',
    ],
    id: '46CitWgnWrvF9t70C2p1Me',
    image: 'https://i.scdn.co/image/8b6e70aafad73059422920cbe528ecbcdff74018',
    name: 'Daughter',
    uri: 'spotify:artist:46CitWgnWrvF9t70C2p1Me',
  },
  {
    genres: [
      'alt-indie rock',
      'garage rock',
      'indie rock',
      'indietronica',
      'modern alternative rock',
      'modern rock',
    ],
    id: '3btzEQD6sugImIHPMRgkwV',
    image: 'https://i.scdn.co/image/f54c5e6b3919940e204406ef6846ee8e197f8217',
    name: 'Wolf Alice',
    uri: 'spotify:artist:3btzEQD6sugImIHPMRgkwV',
  },
];

export const topTracks = [
  {
    album: 'Gone Now',
    artist: 'Bleachers',
    id: '3ySU5vwQB33iGulwcUL9qQ',
    image: 'https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb',
    title: 'Don\'t Take The Money',
    uri: 'spotify:track:3ySU5vwQB33iGulwcUL9qQ',
  },
  {
    album: 'Something To Tell You',
    artist: 'HAIM',
    id: '20QoFTI0fp2dNNexlHDhTe',
    image: 'https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57',
    title: 'Walking Away',
    uri: 'spotify:track:20QoFTI0fp2dNNexlHDhTe',
  },
  {
    album: 'Big Mess',
    artist: 'Grouplove',
    id: '1Tcd9flKIOxXWqR28aDP3U',
    image: 'https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1',
    title: 'Don\'t Stop Making It Happen',
    uri: 'spotify:track:1Tcd9flKIOxXWqR28aDP3U',
  },
];

export const playlist = {
  name: 'Top Tracks',
  description: '',
  image: 'https://mosaic.scdn.co/640/129c6e4a622dc50c0802042780f92631a9dcc14056cc004f3ff47cff0ad8fcf05461730a193d8d8263b1ad734c51e06c7faee7fc95fd3abf0bbb9bbe6f4c08a3f6b73b062d6ea8384add3e3889814448',
  url: 'https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt',
  tracks: topTracks,
};
