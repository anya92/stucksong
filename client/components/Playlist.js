import React from 'react';
import {
  PlaylistContainer,
  PlaylistLinks,
  PlaylistLinkShare,
  PlaylistTracks,
  PlaylistTrackTitle,
  PlaylistTrackInfo,
} from '../styles/playlist';
import PlaylistShare from './PlaylistShare';
import spotifyImage from '../assets/spotify-logo.jpg';

const Playlist = ({ playlist: { name, description, image, tracks, url }, modal, toggleModal }) => (
  <div>
    <h1 style={{ textAlign: 'center', marginBottom: '40px' }}>Your playlist has been successfully created!</h1>
    <PlaylistContainer>
      <div>
        <img src={image || spotifyImage} alt="playlist" />
        <h1>{name}</h1>
        { description && <h4>{description}</h4> }
      </div>
      <div>
        <PlaylistLinks>
          <a href={url} target="_blank" rel="noopener noreferrer">Open in Spotify</a>
          <PlaylistLinkShare>
            Share with your friends:
            <PlaylistShare url={url} name={name} />
          </PlaylistLinkShare>
        </PlaylistLinks>
        <PlaylistTracks>
          { !tracks.length && <div>This playlist is empty</div> }
          {
            tracks.map(({ id, title, artist, album }, i) => (
              <div key={id}>
                <PlaylistTrackTitle><span>{i + 1}.</span> {title}</PlaylistTrackTitle>
                <PlaylistTrackInfo>{artist}, <em>{album}</em></PlaylistTrackInfo>
              </div>
            ))
          }
        </PlaylistTracks>

      </div>
    </PlaylistContainer>
  </div>
);

export default Playlist;
