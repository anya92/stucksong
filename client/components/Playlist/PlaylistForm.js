import React from 'react';
import { Link } from 'react-router-dom';

import {
  PlaylistContainer,
  FormGroup,
  InputGroup,
  InputGroupAddon,
  DangerInfo,
  FormButton,
} from '../../styles/playlist';

const PlaylistForm = ({
  name,
  description,
  numberOfTracks,
  handleChange,
  createPlaylist,
}) => (
  <PlaylistContainer>
    <h1>Create a playlist of your <Link to="/top-tracks">Top&nbsp;Tracks</Link></h1>
    <form onSubmit={e => createPlaylist(e)}>
      <FormGroup>
        <label htmlFor="playlist-name">Name
          <InputGroup>
            <input
              id="playlist-name"
              type="text"
              maxLength="100"
              value={name}
              onChange={e => handleChange('name', e.target.value)}
            />
            <InputGroupAddon>{name.length}/100</InputGroupAddon>
          </InputGroup>
        </label>
      </FormGroup>
      <FormGroup>
        <label htmlFor="playlist-description">Description
          <InputGroup>
            <textarea
              id="playlist-description"
              rows={3}
              maxLength="300"
              placeholder="Give your playlist a description."
              value={description}
              onChange={e => handleChange('description', e.target.value)}
            />
            <InputGroupAddon>
              {description.length}/300
            </InputGroupAddon>
          </InputGroup>
        </label>
      </FormGroup>
      <FormGroup number>
        <label htmlFor="playlist-number-of-tracks">Number of Tracks
          <input
            id="playlist-number-of-tracks"
            type="number"
            className={`${numberOfTracks < 1 || numberOfTracks > 50 ? 'danger' : ''}`}
            min={1}
            max={50}
            value={numberOfTracks}
            onChange={e => handleChange('numberOfTracks', e.target.value)}
          />
          {
            (numberOfTracks < 1 || numberOfTracks > 50)
            && <DangerInfo>Number of tracks must be between 1  and 50.</DangerInfo>
          }
        </label>
      </FormGroup>
      <FormButton>
        <button type="submit">Create</button>
      </FormButton>
    </form>
  </PlaylistContainer>
);

export default PlaylistForm;
