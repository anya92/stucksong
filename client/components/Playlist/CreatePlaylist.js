import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  shape,
  oneOfType,
  bool,
  string,
  number,
  arrayOf,
  func,
} from 'prop-types';
import { createPlaylist } from '../../actions';
import Loadable from '../HOC/Loadable';

import PlaylistForm from './PlaylistForm';
import Loader from '../../styles/loader';

const AsyncError = Loadable({
  loader: () => import('../Error/ErrorComponent'),
});

const AsyncPlaylist = Loadable({
  loader: () => import('./Playlist'),
});

export class CreatePlaylist extends Component {
  state = {
    name: `${this.props.auth.username}'s Top Tracks`,
    description: '',
    numberOfTracks: 25,
  }

  createPlaylist = (e) => {
    e.preventDefault();
    const { name, description, numberOfTracks } = this.state;
    this.props.createPlaylist(name, description, numberOfTracks);
  }

  handleChange = (type, value) => {
    this.setState({ [type]: value });
  }

  render() {
    const { pending, playlist, error } = this.props.playlist;
    if (error) return <AsyncError error={error} />;
    if (pending) return <Loader>Loading...</Loader>;
    if (!playlist) {
      return (
        <PlaylistForm
          name={this.state.name}
          description={this.state.description}
          numberOfTracks={this.state.numberOfTracks}
          handleChange={this.handleChange}
          createPlaylist={this.createPlaylist}
        />
      );
    }
    return (
      <AsyncPlaylist
        playlist={playlist}
      />
    );
  }
}

CreatePlaylist.propTypes = {
  auth: shape({
    username: string,
  }).isRequired,
  playlist: shape({
    pending: bool.isRequired,
    error: oneOfType([bool, string]).isRequired,
    playlist: shape({
      name: string,
      description: string,
      image: string,
      numberOfTracks: number,
      tracks: arrayOf(shape({})),
    }),
  }).isRequired,
  createPlaylist: func.isRequired,
};

function mapStateToProps(state) {
  return {
    auth: state.auth,
    playlist: state.playlist,
  };
}

export default connect(mapStateToProps, { createPlaylist })(CreatePlaylist);
