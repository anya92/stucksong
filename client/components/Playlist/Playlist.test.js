import React from 'react';
import { shallow } from 'enzyme';

import Playlist from './Playlist';
import { playlist } from '../../assets/sampleData';

describe('PLaylist component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Playlist playlist={playlist} />);
    expect(wrapper).toMatchSnapshot();
  });
});
