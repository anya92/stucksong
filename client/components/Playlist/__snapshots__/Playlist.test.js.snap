// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PLaylist component renders correctly 1`] = `
<div>
  <h1
    style={
      Object {
        "marginBottom": "40px",
        "textAlign": "center",
      }
    }
  >
    Your playlist has been successfully created!
  </h1>
  <glamorous(div)>
    <div>
      <img
        alt="playlist"
        src="https://mosaic.scdn.co/640/129c6e4a622dc50c0802042780f92631a9dcc14056cc004f3ff47cff0ad8fcf05461730a193d8d8263b1ad734c51e06c7faee7fc95fd3abf0bbb9bbe6f4c08a3f6b73b062d6ea8384add3e3889814448"
      />
      <h1>
        Top Tracks
      </h1>
    </div>
    <div>
      <glamorous(div)>
        <a
          href="https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt"
          rel="noopener noreferrer"
          target="_blank"
        >
          Open in Spotify
        </a>
        <glamorous(div)>
          Share with your friends:
          <PlaylistShare
            name="Top Tracks"
            url="https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt"
          />
        </glamorous(div)>
      </glamorous(div)>
      <glamorous(div)>
        <div
          key="3ySU5vwQB33iGulwcUL9qQ"
        >
          <glamorous(div)>
            <span>
              1
              .
            </span>
             
            Don't Take The Money
          </glamorous(div)>
          <glamorous(div)>
            Bleachers
            , 
            <em>
              Gone Now
            </em>
          </glamorous(div)>
        </div>
      </glamorous(div)>
    </div>
  </glamorous(div)>
</div>
`;
